src/databricks/labs/remorph/reconcile/compare.py

```diff
@@ -0,0 +1,86 @@
+from pyspark.sql import DataFrame
+from pyspark.sql.functions import col, expr, lit
+
+from databricks.labs.remorph.reconcile.constants import Constants
+from databricks.labs.remorph.reconcile.recon_config import ReconcileOutput
+
+
+def reconcile_data(source: DataFrame, target: DataFrame, join_columns: list[str], report_type: str) -> ReconcileOutput:
+    source_alias = 'src'
+    target_alias = 'tgt'
+    if report_type not in {"data", "all"}:
+        join_columns = Constants.hash_column_name
+    df = source.alias(source_alias).join(other=target.alias(target_alias), on=join_columns, how="full")
+
+    mismatch = _get_mismatch_data(df, source_alias, target_alias) if report_type in {"all", "data"} else None
+    missing_in_src = (
+        df.filter(col(f"{source_alias}.{Constants.hash_column_name}").isNull())
+        .select((join_columns if report_type == "all" else f"{target_alias}.*"))
+        .drop(Constants.hash_column_name)
+    )
+    missing_in_tgt = (
+        df.filter(col(f"{target_alias}.{Constants.hash_column_name}").isNull())
+        .select((join_columns if report_type == "all" else f"{source_alias}.*"))
```

Comment:
```suggestion
        .select((join_columns if report_type == "all" else f"{source_alias}.source_columns"))
```

I will rename to something like this

Reply:
done.